### ISSUE TITLE
Backup fails - The NTFS partition is in an unsafe state - possible Windows 10 shutdown fix?

### ISSUE THREAD

**Issue by PhoenixIV** [redorescue/redorescue#74]
I found that Redo Rescue could not write to my destination disk. When I tried to mount it via the command line I got the error message "The NTFS partition is in an unsafe state".[ I then found the information that on regular shutdown Windows 10 leaves the system in a "hibernated" (invalid) state (askubuntu.com)](https://askubuntu.com/questions/462381/cant-mount-ntfs-drive-the-disk-contains-an-unclean-file-system).

When I shut down Windows with "shutdown /s /t 0" and then performed the backup everything worked fine.

As this seems to be a regular issue with Windows 10, is it possible to apply an automatic fix?

Alternatively: May you implement a troubleshoot dialog that pops up when this issue is detected?

**Comment by zebradots** [comment]
@PhoenixIV: Yes, this is a known issue with the way Windows 10 shuts down. If you determine a Linux-based CLI way to detect or correct this prior to imaging (or better yet, a PR), we would be happy to implement it!

**Comment by GakunGak** [comment]
@zebradots 
Kinda late to the party.

So, basically the hibernation shutdown in Windows 8 and greater is meant to speed up the process of booting, which means saving operating memory data to hiberfil.sys. In doing so, it marks the partition/drive to other systems as [do not touch, hibernated system!]. 
As a precaution [messing with the hibernated system can lead to data loss] and out of respect [do not touch], linux will not access it or may open it in read only mode.

The only workaround is to remove the hiberfil.sys file [which will lead to permanent operating memory data loss!], 
as mentioned here:

https://askubuntu.com/questions/145902/unable-to-mount-windows-ntfs-filesystem-due-to-hibernation

If you need to mount the partition in read-write mode and are not able to or willing to boot into Windows and shut it down completely there is a third option. However, it is not included here because it completely deletes hiberfil.sys and will cause you to lose all unsaved information in the hibernated Windows programs. The following is a quotation from man ntfs-3g about the option that would be used to do this.

remove_hiberfile
              Unlike in case of  read-only  mount,  the  read-write  mount  is
              denied  if  the  NTFS  volume is hibernated. One needs either to
              resume Windows and shutdown it  properly,  or  use  this  option
              which  will  remove  the  Windows hibernation file. Please note,
              this means that the saved Windows  session  will  be  completely
              lost. Use this option under your own responsibility.

So, the command could be invoked [with several warnings why it may be a bad idea to do that] as a method of last resort. Ideally, booting into windows and permanently turning off fast boot is a more ideal approach.

**Comment by PhoenixIV** [comment]
> So, the command could be invoked [with several warnings why it may be a bad idea to do that]

Plus information on how this can be fixed by disabling fast boot or using `shutdown /s /t 0` :)
